istioctl/pkg/waypoint/waypoint.go

```diff
@@ -480,13 +481,31 @@ func deleteWaypoints(cmd *cobra.Command, kubeClient kube.CLIClient, namespace st
 			if gw.Spec.GatewayClassName != constants.WaypointGatewayClassName {
 				continue
 			}
-			names = append(names, gw.Name)
+			nameList = append(nameList, gw.Name)
+		}
+	} else {
+		gws, err := kubeClient.GatewayAPI().GatewayV1().Gateways(namespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, name := range names {
+			contain, gw := waypointContains(gws.Items, name)
```

Comment:
nit: why not combine this with the gateway class validation below? This function is a bit confused to me and only contains a simple forloop. Probably just build a map with valid waypoints, and check if the name exists in the map here.

Reply:
updated